src/base/bittorrent/abstractfilestorage.cpp

```diff
@@ -49,77 +50,60 @@ namespace
     }
 }
 
-void BitTorrent::AbstractFileStorage::renameFile(const QString &oldPath, const QString &newPath)
+void BitTorrent::AbstractFileStorage::renameFile(const Path &oldPath, const Path &newPath)
 {
-    if (!Utils::Fs::isValidFileSystemName(oldPath, true))
-        throw RuntimeError {tr("The old path is invalid: '%1'.").arg(oldPath)};
-    if (!Utils::Fs::isValidFileSystemName(newPath, true))
-        throw RuntimeError {tr("The new path is invalid: '%1'.").arg(newPath)};
-
-    const QString oldFilePath = Utils::Fs::toUniformPath(oldPath);
-    if (oldFilePath.endsWith(QLatin1Char {'/'}))
-        throw RuntimeError {tr("Invalid file path: '%1'.").arg(oldFilePath)};
-
-    const QString newFilePath = Utils::Fs::toUniformPath(newPath);
-    if (newFilePath.endsWith(QLatin1Char {'/'}))
-        throw RuntimeError {tr("Invalid file path: '%1'.").arg(newFilePath)};
-    if (QDir().isAbsolutePath(newFilePath))
-        throw RuntimeError {tr("Absolute path isn't allowed: '%1'.").arg(newFilePath)};
+    if (!oldPath.isValid())
+        throw RuntimeError(tr("The old path is invalid: '%1'.").arg(oldPath.toString()));
```

Comment:
toNative()
I think this error message could be shown to the user.

Reply:
> toNative()
> I think this error message could be shown to the user.

Done.